data/userdataloader.h

```diff
@@ -2,12 +2,18 @@
 #define USERDATALOADER_H
 
 #include <QJsonDocument>
+#include <QJsonObject>
+#include <QFile>
+#include <QDebug>
+#include <QMap>
 
 class UserDataLoader
 {
-    QString user;
 public:
     UserDataLoader(QString);
+    QVariant GetMap(QString);
```

Comment:
Is's a bad name for this method. Rename it in QVariant GetPreference(QString);
also, you need to write a method that returns the whole map (something like QMap<QString,QVariant> *GetSettings() { return map; })